test/CMakeLists.txt

```diff
@@ -58,16 +58,20 @@ endfunction()
 
 add_subdirectory(common)
 
+if (UMF_BUILD_SHARED_LIBRARY)
+    set(LIBS_UMF_SHARED_LIBRARY numa)
```

Comment:
I don't understand this logic - how the fact that umf is a shared lib is connected with adding numa to this variable. Event if this fixes the problem this should not be done this way.

Reply:
`${LIBS_UMF_SHARED_LIBRARY}` is just `numa` now as you can see.
When `libumf` is a static library, `libnuma` is linked with it statically.
When `libumf` is a dynamic library, `libnuma` has to be linked dynamicly with all its targets (tests in this case).

So how should it be done?

Reply:
imagine UMF depends on libA, libB, libC.

If UMF is built as a static library, all dependencies of libumf should be linked with targets that uses libumf. E.g. if ubench uses UMF then ubench should be linked with libA, libB, libC
if UMF is built as a dynamic library, all dependencies are linked directly with libumf.so. E.g. ubench should only links with libumf.

Regarding how it should be implemented with CMake, I am not sure yet.

Reply:
@vinser52 It is seems to be the opposite: libumf is a static library by default now and tests do not need to be linked with libnuma. When you change libumf to a dynamic library the tests need to be linked with libnuma ...

Reply:
> When you change libumf to a dynamic library the tests need to be linked with libnuma

Why? Who does actually need the libnuma dependency? libumf uses libnuma, right?

Reply:
You are right. I have not cleaned up the previous build and everything got mixed ...
It is fixed now. Please review again.

Reply:
Neither tests nor benchmarks should link with UMF dependencies (regardless of whether UMF is built as static or shared library) unless they use the libraries themselves. See: https://github.com/oneapi-src/unified-memory-framework/pull/89